### PR TITLE
Page Layout Tab Structure

### LINKED ISSUE #5079: Page Layouts: Create Tab and Layout Structure for Page Layouts in Workspace
## Feature Description

Behind a feature flag create the tab with icon for Page Layouts with the header and containers. This does not include the page layouts themselves or the search field.

https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/Stories---Post-Stable?node-id=2%3A98474

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

1. When the flag is true, opening the editor should reveal a tab that will contain the page layouts
2. It should have the header portion with the page layout filter pills (that mirror text sets)

### PR DESCRIPTION
**Pull request by zachhale**
## Summary

Adds page layouts tab structure hidden behind a feature flag that are based on templates.

* Adds template loading function to `useAPI`
* This includes pills, but not the search field
* Page layouts (template pages) are rendered as empty boxes, but are hooked up with `react-virtual`
* Selecting a pill filters the page set

[ticket](https://app.zenhub.com/workspaces/web-stories-5e94d3aced449034e1e9b226/issues/google/web-stories-wp/5079)

## User-facing changes

Hidden behind a feature flag for "Page layouts tab"

image

## Testing Instructions

* To view tab, enable "Page layouts tab" in Experiments

---

Fixes #5079

### CODE REVIEW

File: assets/src/edit-story/components/library/panes/pageLayouts/pageLayouts.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        const indexes = [firstColumnIndex, firstColumnIndex + 1];
+        return (
+          <PageLayoutsRow
+            key={virtualRow.index}
+            height={virtualRow.size}
+            translateY={virtualRow.start}
+          >
+            {indexes.map((index) => {
+              const page = pages[index];
+              return page ? <PageLayout key={index} page={page} /> : null;
```
Comment (carloskelly13) [inline assets/src/edit-story/components/library/panes/pageLayouts/pageLayouts.js]: Is there a way to get an identifier here over index? Even if we attach it at the source when we pull from the data, like `uuid4()`?

Comment (zachhale) [reply assets/src/edit-story/components/library/panes/pageLayouts/pageLayouts.js]: Ah yeah, I can pull in the page id!

File: assets/src/edit-story/components/library/panes/pageLayouts/pageLayouts.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+          </PageLayoutsRow>
+        );
+      })}
+    </PageLayoutsContainer>
+  );
+}
+
+PageLayouts.propTypes = {
+  parentRef: PropTypes.object,
+  pages: PropTypes.array,
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/library/panes/pageLayouts/pageLayouts.js]: Are there any benefits in detailing what we should expect in these propTypes other than an object and an array?

File: assets/src/edit-story/icons/page_layouts.svg [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+<path fill-rule="evenodd" clip-rule="evenodd" d="M1.5 0C0.671573 0 0 0.671573 0 1.5V7.5C0 8.32843 0.671573 9 1.5 9H5.5C6.32843 9 7 8.32843 7 7.5V1.5C7 0.671573 6.32843 0 5.5 0H1.5ZM1 1.5C1 1.22386 1.22386 1 1.5 1H5.5C5.77614 1 6 1.22386 6 1.5V7.5C6 7.77614 5.77614 8 5.5 8H1.5C1.22386 8 1 7.77614 1 7.5V1.5ZM1.5 11C0.671573 11 0 11.6716 0 12.5V14.5C0 15.3284 0.671573 16 1.5 16H5.5C6.32843 16 7 15.3284 7 14.5V12.5C7 11.6716 6.32843 11 5.5 11H1.5ZM1 12.5C1 12.2239 1.22386 12 1.5 12H5.5C5.77614 12 6 12.2239 6 12.5V14.5C6 14.7761 5.77614 15 5.5 15H1.5C1.22386 15 1 14.7761 1 14.5V12.5ZM9 1.5C9 0.671573 9.67157 0 10.5 0H14.5C15.3284 0 16 0.671573 16 1.5V4.5C16 5.32843 15.3284 6 14.5 6H10.5C9.67157 6 9 5.32843 9 4.5V1.5ZM10.5 1C10.2239 1 10 1.22386 10 1.5V4.5C10 4.77614 10.2239 5 10.5 5H14.5C14.7761 5 15 4.77614 15 4.5V1.5C15 1.22386 14.7761 1 14.5 1H10.5ZM10.5 8C9.67157 8 9 8.67157 9 9.5V14.5C9 15.3284 9.67157 16 10.5 16H14.5C15.3284 16 16 15.3284 16 14.5V9.5C16 8.67157 15.3284 8 14.5 8H10.5ZM10 9.5C10 9.22386 10.2239 9 10.5 9H14.5C14.7761 9 15 9.22386 15 9.5V14.5C15 14.7761 14.7761 15 14.5 15H10.5C10.2239 15 10 14.7761 10 14.5V9.5Z" fill="#E4E5E6"/>
+</svg>
```
Comment (BrittanyIRL) [inline assets/src/edit-story/icons/page_layouts.svg]: If you want, this is also in the new design system icons. Em referenced some alert icons in her prepubish checklist. this one would be `const { Layout } from '../../blah/.../design-system/icons/editor/'`

**Review by BrittanyIRL — APPROVED** [review]
This all makes sense and renders as expected! 
I did notice a bug, i'm not sure it needs to be fixed here because I think it relates more to the tab structure as a whole. 

If the browser window isn't big enough to render all of the tab titles you'll never be able to get to the page layouts because the tab titles don't allow you to scroll through options, nor is there a way to know that there's sections hidden.  
layouts tab disappears

File: assets/src/edit-story/components/library/panes/pageLayouts/pageLayouts.js (hunk 2 above)
Comment (zachhale) [reply assets/src/edit-story/components/library/panes/pageLayouts/pageLayouts.js]: I'll add the basics, sure.

File: assets/src/edit-story/icons/page_layouts.svg (hunk 3 above)
Comment (zachhale) [reply assets/src/edit-story/icons/page_layouts.svg]: Is that cool to do so instead of putting it in `edit-story`?

**Comment by zachhale** [comment]
@BrittanyIRL The small browser issue is going to show itself on the other tabs too with manually defined widths for items in the tabs. I think we should punt on solving the small browser issue because of that.

File: assets/src/edit-story/icons/page_layouts.svg (hunk 3 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/icons/page_layouts.svg]: i think so! There's no added dependencies and it's helping avoid duplication as new icons are added.

**Pull request merged**